helix-term/src/commands.rs

```diff
@@ -1541,6 +1573,18 @@ fn delete_selection_impl(reg: &mut Register, doc: &mut Document, view_id: ViewId
     doc.apply(&transaction, view_id);
 }
 
+#[inline]
+fn delete_selection_insert_mode(cx: &mut Context, selection: &Selection) {
+    let (view, doc) = current!(cx.editor);
+    let view_id = view.id;
+
+    // then delete
+    let transaction = Transaction::change_by_selection(doc.text(), selection, |range| {
+        (range.from(), range.to(), None)
+    });
+    doc.apply(&transaction, view_id);
+}
+
```

Comment:
I would just inline this function everywhere, but if you want to avoid that then maybe use `delete_selection_insert_mode(doc: &mut Document, view: &View, selection: &Selection)`, that way we avoid looking up `current!` again.

Reply:
Good point, updated accordingly.